File: samples/client/wordnik/java/src/main/java/io/swagger/client/api/WordListsApi.java
package io.swagger.client.api;

import com.sun.jersey.multipart.FormDataMultiPart;
import io.swagger.client.ApiException;
import io.swagger.client.ApiInvoker;
import io.swagger.client.model.WordList;

import java.util.HashMap;
import java.util.Map;

public class WordListsApi {
    String basePath = "https://api.wordnik.com/v4";
    ApiInvoker apiInvoker = ApiInvoker.getInstance();

    public ApiInvoker getInvoker() {
        return apiInvoker;
    }

    public String getBasePath() {
        return basePath;
    }

    public void setBasePath(String basePath) {
        this.basePath = basePath;
    }

    public WordList createWordList(WordList body, String auth_token) throws ApiException {
        Object postBody = body;


        // create path and map variables
        String path = "/wordLists.json".replaceAll("\\{format\\}", "json");

        // query params
        Map<String, String> queryParams = new HashMap<String, String>();
        Map<String, String> headerParams = new HashMap<String, String>();
        Map<String, String> formParams = new HashMap<String, String>();


        headerParams.put("auth_token", auth_token);

        String[] contentTypes = {

        };

        String contentType = contentTypes.length > 0 ? contentTypes[0] : "application/json";

        if (contentType.startsWith("multipart/form-data")) {
            boolean hasFields = false;
            FormDataMultiPart mp = new FormDataMultiPart();

            if (hasFields) {
                postBody = mp;
            }
        } else {

        }

        try {
            String response = apiInvoker.invokeAPI(basePath, path, "POST", queryParams, postBody, headerParams, formParams, contentType);
            if (response != null) {
                return (WordList) ApiInvoker.deserialize(response, "", WordList.class);
            } else {
                return null;
            }
        } catch (ApiException ex) {
            if (ex.getCode() == 404) {
                return null;
            } else {
                throw ex;
            }
        }
    }

}

File: samples/client/wordnik/scala/src/main/scala/io/swagger/client/api/WordListsApi.scala
package io.swagger.client.api

import java.io.File

import scala.collection.mutable.HashMap

class WordListsApi(val defBasePath: String = "https://api.wordnik.com/v4",
                   defApiInvoker: ApiInvoker = ApiInvoker) {
  var basePath = defBasePath
  var apiInvoker = defApiInvoker

  def addHeader(key: String, value: String) = apiInvoker.defaultHeaders += key -> value


  def createWordList(body: WordList, auth_token: String): Option[WordList] = {
    // create path and map variables
    val path = "/wordLists.json".replaceAll("\\{format\\}", "json")


    val contentType = {
      if (body != null && body.isInstanceOf[File])
        "multipart/form-data"
      else "application/json"


    }

    // query params
    val queryParams = new HashMap[String, String]
    val headerParams = new HashMap[String, String]





    headerParams += "auth_token" -> auth_token


    try {
      apiInvoker.invokeApi(basePath, path, "POST", queryParams.toMap, body, headerParams.toMap, contentType) match {
        case s: String =>
          Some(ApiInvoker.deserialize(s, "", classOf[WordList]).asInstanceOf[WordList])

        case _ => None
      }
    } catch {
      case ex: ApiException if ex.code == 404 => None
      case ex: ApiException => throw ex
    }
  }

}
